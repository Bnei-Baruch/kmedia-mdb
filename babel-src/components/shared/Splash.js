'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ErrorSplash = exports.FrownSplash = exports.LoadingSplash = exports.Splash = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _semanticUiReact = require('semantic-ui-react');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Splash = exports.Splash = function Splash(props) {
  var text = props.text,
      subtext = props.subtext,
      icon = props.icon,
      color = props.color,
      isLoading = props.isLoading;


  return _react2.default.createElement(
    _semanticUiReact.Header,
    { as: 'h2', icon: true, textAlign: 'center' },
    _react2.default.createElement(_semanticUiReact.Icon, { name: icon, loading: isLoading, color: color }),
    _react2.default.createElement(
      _semanticUiReact.Header.Content,
      null,
      text,
      subtext ? _react2.default.createElement(
        _semanticUiReact.Header.Subheader,
        null,
        subtext
      ) : null
    )
  );
};

Splash.propTypes = {
  icon: _propTypes2.default.string.isRequired,
  text: _propTypes2.default.node.isRequired,
  subtext: _propTypes2.default.node,
  color: _propTypes2.default.string,
  isLoading: _propTypes2.default.bool
};

Splash.defaultProps = {
  text: '',
  subtext: null,
  color: 'black',
  isLoading: false
};

var LoadingSplash = exports.LoadingSplash = function LoadingSplash(props) {
  return _react2.default.createElement(Splash, Object.assign({ isLoading: true, icon: 'spinner' }, props));
};
var FrownSplash = exports.FrownSplash = function FrownSplash(props) {
  return _react2.default.createElement(Splash, Object.assign({ icon: 'frown', color: 'orange' }, props));
};
var ErrorSplash = exports.ErrorSplash = function ErrorSplash(props) {
  return _react2.default.createElement(Splash, Object.assign({ icon: 'warning sign', color: 'red' }, props));
};