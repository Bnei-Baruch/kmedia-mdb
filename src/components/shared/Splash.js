import React from 'react';
import PropTypes from 'prop-types';
import { Header, Icon } from 'semantic-ui-react';

export const Splash = (props) => {
  const { text, subtext, icon, color, isLoading } = props;

  return (
    <Header as="h2" icon textAlign="center">
      <Icon name={icon} loading={isLoading} color={color} />
      <Header.Content>
        {text}
        {subtext ? <Header.Subheader>{subtext}</Header.Subheader> : null}
      </Header.Content>
    </Header>
  );
};

Splash.propTypes = {
  icon: PropTypes.string.isRequired,
  text: PropTypes.node.isRequired,
  subtext: PropTypes.node,
  color: PropTypes.string,
  isLoading: PropTypes.bool,
};

Splash.defaultProps = {
  text: '',
  subtext: null,
  color: 'black',
  isLoading: false,
};

export const LoadingSplash = props => (<Splash isLoading icon="spinner" {...props} />);
export const FrownSplash   = props => (<Splash icon="frown" color="orange" {...props} />);
export const ErrorSplash   = props => (<Splash icon="warning sign" color="red" {...props} />);
