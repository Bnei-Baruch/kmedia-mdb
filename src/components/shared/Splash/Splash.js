import React from 'react';
import PropTypes from 'prop-types';
import { Header, Icon } from 'semantic-ui-react';

export const Splash = props => {
  const { text, subtext = null, icon, color = 'black', isLoading = false, item = null } = props;

  return (
    <div className="splash">
      <Header icon as="h2" textAlign="center">
        <Icon name={icon} loading={isLoading} color={color} />
        <Header.Content>
          {text}
          {subtext ? <Header.Subheader data-testid='header.subHeader'>{subtext}</Header.Subheader> : null}
          {item ? <Header.Subheader>{item}</Header.Subheader> : null}
        </Header.Content>
      </Header>
    </div>
  );
};

Splash.propTypes = {
  icon: PropTypes.string.isRequired,
  text: PropTypes.node.isRequired,
  subtext: PropTypes.node,
  color: PropTypes.string,
  isLoading: PropTypes.bool,
  item: PropTypes.func,
};

export const LoadingSplash = props => (<Splash isLoading icon="circle notch" color="blue" {...props} />);
export const FrownSplash   = props => (<Splash icon="frown" color="orange" {...props} />);
export const ErrorSplash   = props => (<Splash icon="warning sign" color="red" {...props} />);
