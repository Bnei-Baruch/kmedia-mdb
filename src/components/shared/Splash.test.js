import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallowToJson } from 'enzyme-to-json';

import { Header, Icon } from 'semantic-ui-react';

import { Splash } from './Splash';

configure({ adapter: new Adapter() });

describe('Splash', () => {
  const shallowSplashRender = props => shallow(<Splash {...props} />);
  const mountedSplashRender = props => mount(<Splash {...props} />);

  let shallowLockScreen;
  const shallowSplash = (props) => {
    if (!shallowLockScreen) {
      shallowLockScreen = shallowSplashRender(props);
    }
    return shallowLockScreen;
  };
  let mountedLockScreen;
  const mountedSplash = (props) => {
    if (!mountedLockScreen) {
      mountedLockScreen = mountedSplashRender(props);
    }
    return mountedLockScreen;
  };

  it('should render correctly', () => {
    const output = shallowSplash({ icon: 'spinner', text: 'Still loading...' });
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('always render a Header', () => {
    const headers = shallowSplash({ icon: '', text: '' }).find(Header);
    expect(headers.length).toBeGreaterThan(0);
  });

  describe('the rendered Header', () => {
    it('contains everything else that gets rendered', () => {
      const headers = shallowSplash({ icon: '', text: '' }).find(Header);

      // When using .find, enzyme arranges the nodes in order such
      // that the outermost node is first in the list. So we can
      // use .first() to get the outermost div.
      const wrappingHeader = headers.first();

      // Enzyme omits the outermost node when using the .children()
      // method on splash(). This is annoying, but we can use it
      // to verify that wrappingHeader contains everything else this
      // component renders.
      expect(wrappingHeader.children()).toEqual(shallowSplash({ icon: '', text: '' }).children());
    });

    it('always renders an Icon', () => {
      expect(shallowSplash({ icon: '', text: '' }).find(Icon)).toHaveLength(1);
    });

    it('always renders an Header.Content', () => {
      const header = shallowSplash({ icon: '', text: '' }).find(Header.Content);
      expect(header).toHaveLength(1);
      it('Header.Content does not receive any props', () => {
        expect(Object.keys(header.props())).toHaveLength(0);
      });
    });
  });

  describe('when subtext is not supplied', () => {
    it('does not renders Header.Subheader', () => {
      const subheader = shallowSplash({ icon: '', text: '', subtext: 'subtext prop' }).find(Header.Subheader);
      expect(subheader).toHaveLength(0);
    });
  });

  describe('when subtext is supplied', () => {
    let subheader;
    const subtext = 'subtext prop';

    beforeEach(() => {
      subheader = mountedSplash({ icon: '', text: '', subtext }).find(Header.Subheader).first();
    });

    it('renders Header.Subheader', () => {
      expect(subheader).toHaveLength(1);
    });
    it('renders Header.Subheader with this subtext', () => {
      expect(subheader.props().children).toBe(subtext);
    });
  });
});

// https://medium.freecodecamp.org/the-right-way-to-test-react-components-548a4736ab22
//
// The concerns of React component:
// * What do I do with the props I receive?
// * What components do I render? What do I pass to those components?
// * Do I ever keep anything in state? If so, do I invalidate it when receiving new props? When do I update state?
// * If a user interacts with me or a child component calls a callback I passed to it, what do I do?
// * Does anything happen when I’m mounted? When I’m unmounted?
//
// Should we test?
// * Will the test have to duplicate exactly the application code? This will make it brittle.
// * Will making assertions in the test duplicate any behavior that is already covered by (and the responsibility of) library code?
// * From an outsider’s perspective, is this detail important, or is it only an internal concern?
//   Can the effect of this internal detail be described using only the component’s public API?
