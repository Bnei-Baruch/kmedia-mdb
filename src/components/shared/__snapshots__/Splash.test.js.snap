// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should render correctly 1`] = `
<Header
  as="h2"
  icon={true}
  textAlign="center"
>
  <Icon
    as="i"
    color="black"
    loading={false}
    name="spinner"
  />
  <HeaderContent>
    Still loading...
  </HeaderContent>
</Header>
`;
